dataavailability/eigenda/client.go

```diff
@@ -0,0 +1,81 @@
+package eigenda
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// ErrNotFound is returned when the server could not find the input.
+var ErrNotFound = errors.New("not found")
+
+// ErrInvalidInput is returned when the input is not valid for posting to the DA storage.
+var ErrInvalidInput = errors.New("invalid input")
+
+// DAClient is an HTTP client to communicate with a DA storage service.
+// It creates commitments and retrieves input data + verifies if needed.
+type DAClient struct {
+	url string
+}
+
+func NewDAClient(url string) *DAClient {
```

Comment:
Might be nice to define an interface for this so it can be mocked out. Not strictly necessary though.